.github/workflows/coding-standard-baseline.yml

```diff
@@ -0,0 +1,67 @@
+name: M2 Coding Testing
+
+on:
+  push:
+    branches: [ 2.4-develop, develop, codingstandard_baseline_integration ]
+  pull_request:
+    branches: [ 2.4-develop, develop, codingstandard_baseline_integration ]
+  workflow_call:
+    inputs:
+      php_version:
+        type: string
+        required: true
+        default: "8.1"
+        description: "PHP version used to do the coding standard check."
+
+      composer_version:
+        type: string
+        required: true
+        default: "2"
+        description: "The version of composer to use."
+
+      path:
+        type: string
+        required: true
+        default: 'app/code'
+        description: "The directory (relative to the project root) in which the coding standard will be checked. Used when the event is not a pull request."
+
+      version:
+        type: string
+        required: false
+        description: "The version of the coding standard to use. If not provided, will use the latest version."
+
+      severity:
+        type: string
+        required: false
+        default: "8"
+        description: "The minimum severity required to display an error or warning (default: 5)"
+
+      warning_severity:
+        type: string
+        required: false
+        default: "8"
+        description: "The minimum severity required to display a warning"
+
+      error_severity:
+        type: string
+        required: false
+        default: "8"
+        description: "The minimum severity required to display an error"
+
+permissions:
+  contents: read
+
+jobs:
+  coding-standard:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: mage-os/github-actions/coding-standard-baseline@main
+        with:
+          php_version: "8.1" # Optional, will be used for Php version
```

Comment:
Is the PHP version even important for phpcs? :thinking: If it is, we should probably test with all supported versions. If it is not, we could probably remove the input altogether?

Reply:
we have to test with php 8.1 and 8.2 both versions